app/services/remove_status_service.rb

```diff
@@ -95,7 +95,7 @@ def remove_from_remote_reach
   end
 
   def signed_activity_json
-    @signed_activity_json ||= Oj.dump(serialize_payload(@status, @status.reblog? ? ActivityPub::UndoAnnounceSerializer : ActivityPub::DeleteSerializer, signer: @account))
+    @signed_activity_json ||= Oj.dump(serialize_payload(@status, @status.reblog? ? ActivityPub::UndoAnnounceSerializer : ActivityPub::DeleteSerializer, signer: @account, always_sign: true))
```

Comment:
Need to check the status isn't inlined in either `Undo { Announce }` nor `Delete { Status }`. I think `Undo { Announce }` could leak the status on self-boosts.

Reply:
Since `always_sign` does not override `Status#sign?`, this is safe, and private self-unboosts would not get signed, but private self-unboosts do indeed include the original status, addressed in #17516